### PR TITLE
feat: Update transaction controller to v8.0.1

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
 This PR aims to update the transaction controller to version 8.0.1.
 
 The Approval Controller was also updated on this PR to the version of 3.5.0 and approval controller patched was updated.

All the changes are present in this core branch for easy review: patch/mobile-transaction-controller-8-0-1

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

transactions: https://recordit.co/mpcRkPQElC
sign: http://recordit.co/yPI78aEozu

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by sleepytanya** [comment]
QA [build](https://app.bitrise.io/build/519fb626-d99d-4fe2-81cb-41488828bb95?tab=artifacts), physical device Samsung s24+, Sepolia, Linea Goerli:

 - [ ] Transactions - send native token origin MM
 - [ ]  Transactions - send native token origin dapp
 - [ ]  Transactions - send ERC20 token origin MM
 - [ ]  Transactions - send ERC20 token origin dapp
 - [ ]  Transactions - send ERC721 token origin MM
 - [ ] Transactions - send ERC721 token origin dapp
 - [ ] Transactions - speed up transaction
 - [ ]  Transactions - cancel transaction
 - [ ] Tokens - import ERC20 token origin MM
 - [ ] Tokens - import ERC20 token origin dapp
 - [ ] Tokens - import ERC721 token origin MM (NFT image is not displayed, not related to current PR)
 - [ ] Tokens - import ERC721 token origin dapp ('watch NFT', 'watch all NFTs' are not responsive, not related to current branch)
 - [ ] Tokens - import ERC1155 token origin MM
 - [ ] Tokens - import ERC1155 token origin dapp
 - [ ] Tokens - approve ERC1155 token
 - [ ] Tokens - approve ERC20 token
 - [ ] Tokens - approve ERC721 token
 - [ ] ENS - name resolution (works on mainnet only, the same way it works on extension)
 - [ ] Gas fee - EIP-1559 gas
 - [ ] Gas fee - legacy gas
 - [ ] Sign - eth sign (Blockaid alert)
 - [ ] Sign - personal sign
 - [ ] Sign - sign in with ethereum
 - [ ] Sign - sign typed with data
 - [ ] Permissions - connecting and disconnecting from a dapp
 - [ ] Swap - swap eth
 - [ ] Incoming transactions history
 - [ ] Custom nonce
 - [ ] Cancel transaction using custom nonce
 - [ ] Stuck multiple transactions (cancel / speed up)
 - [ ] Malformed signatures (alert for 'invalid type' only)
 - [ ] PPOM

File: patches/@metamask+transaction-controller+8.0.1.patch [hunk 1]
```diff
@@ -0,0 +1,2747 @@
...
+--- /dev/null
++++ b/node_modules/@metamask/transaction-controller/dist/.patch.txt
+@@ -0,0 +1,7 @@
++PATCH GENERATED FROM MetaMask/core branch: refactor/transaction-controller-patch-mobile-5
++This patch backports various transaction controller features from the main branch of MetaMask/core
++Steps to update patch:
++* Create a new core branch from: patch/mobile-transaction-controller-7-1-0
++* Run "yarn build" in the core monorepo
++* Run "yarn patch:tx <core-directory>" in the mobile repo
++* Once the new patch is merged, add your changes to: patch/mobile-transaction-controller-6-1-0
```
Comment (matthewwalsh0) [inline patches/@metamask+transaction-controller+8.0.1.patch:13]: Minor, but is it worth updating this and the above branches for clarity? Line 7, 10, and 13.

Plus should we update the `patch:tx` script to reference the new branch?

File: patches/@metamask+approval-controller+3.5.0.patch [hunk 2]
```diff
@@ -0,0 +1,94 @@
+diff --git a/node_modules/@metamask/approval-controller/dist/ApprovalController.d.ts b/node_modules/@metamask/approval-controller/dist/ApprovalController.d.ts
+index 32f61cb..4e4d7a1 100644
+--- a/node_modules/@metamask/approval-controller/dist/ApprovalController.d.ts
++++ b/node_modules/@metamask/approval-controller/dist/ApprovalController.d.ts
+@@ -99,6 +99,12 @@ export declare type AcceptOptions = {
+      * If false or unspecified, the promise will resolve immediately.
+      */
+     waitForResult?: boolean;
++    /**
```
Comment (matthewwalsh0) [inline patches/@metamask+approval-controller+3.5.0.patch:9]: For the sake of audit, I'll create the corresponding core branch now so we can track these changes.

Comment (tommasini) [reply patches/@metamask+approval-controller+3.5.0.patch:9]: Thanks!!!

File: patches/@metamask+transaction-controller+8.0.1.patch (hunk 1 above)
Comment (tommasini) [reply patches/@metamask+transaction-controller+8.0.1.patch:13]: Thanks!! Changes pushed!

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**